Quaternion refactor

mschachter opened the pull request:
Refactored quaternion code to use MuJoCo quaternion functions instead of pyquaternion.	

- Refactored redundant functions in sawyer_xyz/sawyer_*	tasks into sawyer_xyz/base.py.
- Swapped pyquaternion functionality in env_utils.py to use MuJoCo quaternion functions.
- Created unit test (test_quat.py) to demonstrate parity between MuJoCo	functions and pyquaternion functions.
- Removed pyquaternion dependency in setup.py from production, added to dev.
- Removed numpy-stl dependency in setup.py.
- metaworld/envs/assets directory was not copied during installation in Linux or Mac. Added MANIFEST.in file to include assets on installation, and added install_package_data=True to setup.py to fix this.

zhanpenghe commented on MANIFEST.in:
```diff
@@ -0,0 +1 @@
+recursive-include metaworld/envs/assets *
```
Can you add a newline after this please?

mschachter replied:
Sure, I'll wait on @tianheyu927 's review, and then commit any changes that are required.

zhanpenghe commented on metaworld/envs/mujoco/sawyer_xyz/sawyer_disassemble_peg.py:
```diff
@@ -186,21 +186,7 @@ def _set_goal_marker(self, goal):
-        qvel[9:15] = 0
-        self.set_state(qpos, qvel)
-
-    def _set_obj_xyz(self, pos):
```
Hmm I am wondering whether if we should remove this. I think we kinda have different `_set_obj_xyz` for different environments, especially that we have two tasks using two objects.

zhanpenghe replied:
But, I think it is good if we can remove this since these codes are copied a lot of times across this codebase

mschachter replied:
Hi @zhanpenghe ! Thank you for the quick review. I used find-and-replace to refactor the functions, they were identical across a few dozen files. In my opinion, if some of the environments need a different implementation of _set_obj_xyz, they can override the function in the implementation.

mschachter commented:
Amicable ping @tianheyu927 , would it be possible for you to review this PR? Thank you!!

tianheyu927 commented:
LGTM

ryanjulian commented on tests/metaworld/envs/test_quat.py:
```diff
@@ -0,0 +1,73 @@
+        q = env_util.zangle_to_quat(ang1)
+        assert_close(exp_q, q)
+
+test_quaternions()
```
this shouldn't be necessary with pytest (it should collect and run the test by virtue of its name (i.e `test_foo`)

mschachter replied:
Ok, will change, thank you!

ryanjulian commented:
FYI your Travis build is failing because we don't share the MuJoCo key with fork PRs (for security reasons).

Please ping here when it's ready to merge, and I'll pull it into a branch here so Travis can run.

mschachter commented:
@ryanjulian @zhanpenghe I addressed your comments on my branch. I think it's ready to be tested on on Travis and merged. Thanks, let me know if there's anything more I can do!

zhanpenghe commented:
Thank you very much! @mschachter
